src/main/java/com/game/Battleship/controllers/BoardController.java

```diff
@@ -1,17 +1,46 @@
 package com.game.Battleship.controllers;
-
 import com.game.Battleship.models.Board;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.game.Battleship.models.Cell;
+import com.game.Battleship.models.Game;
+import com.game.Battleship.models.Ship;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+import java.util.Map;
 
 @RestController
 public class BoardController {
 
-    @GetMapping("/board")
-    public Board getBoard(){
+    @Autowired
+    private Game game;
+
+    @GetMapping("/boards")
+    public List<Board> getAllBoards() {
 
-        Board board = new Board();
-        return board;
+        return game.getAllBoards();
     }
 
-}
+    //json object example: {"row": 0, "column": 1, "shipType": "Cruiser", "shipSize": 3, "isHorizontal": false }
+    @PatchMapping(value = "/boards/place")
+    public Board placeShip(@RequestBody Map<String, Object> changes) {
+
+        Integer row = (Integer) changes.get("row");
+        Integer column = (Integer) changes.get("column");
+        String stringShipType = (String) changes.get("shipType");
+        Cell.Type shipType = Cell.Type.valueOf(stringShipType.toUpperCase());
+        Integer shipSize = (Integer) changes.get("shipSize");
+        Boolean isHorizontal = (Boolean) changes.get("isHorizontal");
```

Comment:
Not feedback but more of a question for my learning - but does Spring Boot require that you typecast the values that you send in the request body?

Reply:
To be honest, I'm not sure if this is a Spring Boot thing but  it gives me when I sign the value  and IntilliJ suggested to typecast the values

Reply:
Ah interesting, good to know!